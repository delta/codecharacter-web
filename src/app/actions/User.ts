import * as UserInterfaces from 'app/types/User';
import { action } from 'typesafe-actions';

export namespace UserActions {
  export enum Type {
    ACTIVATE_USER = 'ACTIVATE_USER',
    LOGIN = 'LOGIN',
    LOGOUT = 'LOGOUT',
    REGISTER = 'REGISTER',
    GET_USER_DETAILS = 'GET_USER_DETAILS',
    GET_USER_QUESTSTATUS = 'GET_USER_QUESTSTATUS',
    UPDATE_USER_QUESTSTATUS = 'UPDATE_USER_QUESTSTATUS',
    SET_CURRENT_LEVEL = 'SET_CURRENT_LEVEL',
    EDIT_USER_PROFILE = 'EDIT_USER_PROFILE',
    EDIT_USER_PASSWORD = 'EDIT_USER_PASSWORD',
    UPDATE_ERROR_MESSAGE = 'UPDATE_ERROR_MESSAGE',
    UPDATE_NOTIFICATION = 'UPDATE_NOTIFICATION',
    UPDATE_USER_DETAILS = 'UPDATE_USER_DETAILS',
    UPDATE_SOCKET_MESSAGE = 'UPDATE_SOCKET_MESSAGE',
    CHECK_EMAIL_EXISTS = 'CHECK_EMAIL_EXISTS',
    CHECK_USERNAME_EXISTS = 'CHECK_USERNAME_EXISTS',
    TOGGLE_USER_PROFILE_MODAL = 'TOGGLE_USER_PROFILE_MODAL',
    TOGGLE_IS_NOTIFICATION_PRESENT = 'TOGGLE_IS_NOTIFICATION_PRESENT',
    TOGGLE_IS_SOCKET_PRESENT = 'TOGGLE_IS_SOCKET_PRESENT',
    RESET_USER_STATE = 'RESET_USER_STATE',
    RESET_APP_STATE = 'RESET_APP_STATE',
    SET_IS_AUTHENTICATION_OPEN = 'SET_IS_AUTHENTICATION_OPEN',
    SET_IS_LOGIN_LOADING = 'SET_IS_LOGIN_LOADING',
    OPEN_STORY_MODE_MODAL = 'OPEN_STORY_MODE_MODAL',
    CLOSE_STORY_MODE_MODAL = 'CLOSE_STORY_MODE_MODAL',
  }

  export const activateUser = (authToken: string, userId: number) =>
    action(Type.ACTIVATE_USER, {
      authToken,
      userId,
    });

  export const login = (email: string, password: string) =>
    action(Type.LOGIN, {
      email,
      password,
    });

  export const logout = () => action(Type.LOGOUT);

  export const register = (registerDetails: UserInterfaces.Register) =>
    action(Type.REGISTER, { registerDetails });

  export const updateErrorMessage = (errorMessage: string) =>
    action(Type.UPDATE_ERROR_MESSAGE, { errorMessage });

  interface UserDetails {
    isLoggedIn: boolean;
    isFirstLogin?: boolean;
    userId?: number;
    avatar?: string;
    college?: string;
    userType?: UserInterfaces.UserType;
    fullName?: string;
    username?: string;
    email?: string;
    errorMessage?: string;
    country?: string;
    isUserProfileModalOpen?: boolean;
  }

  export const updateUserDetails = (userDetails: UserDetails) =>
    action(Type.UPDATE_USER_DETAILS, { userDetails });

  export const getUserDetails = () => action(Type.GET_USER_DETAILS);

  export const getQuestStatus = () => action(Type.GET_USER_QUESTSTATUS);

  export const updateQuestStatus = (ratings: object) =>
    action(Type.UPDATE_USER_QUESTSTATUS, { ratings });

  export const setCurrentLevel = (level: number, stars: number) =>
    action(Type.SET_CURRENT_LEVEL, { level, stars });

  export const editUserProfile = (editUserDetails: UserInterfaces.EditUserDetails) =>
    action(Type.EDIT_USER_PROFILE, { editUserDetails });

  export const editUserPassword = (editUserPasswordDetails: UserInterfaces.EditUserPassword) =>
    action(Type.EDIT_USER_PASSWORD, { editUserPasswordDetails });

  export const checkEmailExists = (email: string) => action(Type.CHECK_EMAIL_EXISTS, { email });

  export const checkUsernameExists = (username: string) =>
    action(Type.CHECK_USERNAME_EXISTS, { username });

  export const toggleIsSocketPresent = () => action(Type.TOGGLE_IS_SOCKET_PRESENT);

  export const updateSocketMessage = (socketMessage: string) =>
    action(Type.UPDATE_SOCKET_MESSAGE, { socketMessage });

  export const toggleIsNotificationPresent = () => action(Type.TOGGLE_IS_NOTIFICATION_PRESENT);

  export const updateNotification = (notification: string) =>
    action(Type.UPDATE_NOTIFICATION, { notification });

  export const toggleUserProfileModal = (isUserProfileModalOpen: boolean) =>
    action(Type.TOGGLE_USER_PROFILE_MODAL, { isUserProfileModalOpen });

  export const resetUserState = () => action(Type.RESET_USER_STATE);

  export const resetAppState = () => action(Type.RESET_APP_STATE);

  export const setIsAuthenticationOpen = (isAuthenticationOpen: boolean) =>
    action(Type.SET_IS_AUTHENTICATION_OPEN, { isAuthenticationOpen });

  export const setIsLoginLoading = (isLoginLoading: boolean) =>
    action(Type.SET_IS_LOGIN_LOADING, { isLoginLoading });

  export const openStoryModeModal = (level: number) =>
    action(Type.OPEN_STORY_MODE_MODAL, { level });

  export const closeStoryModeModal = () => action(Type.CLOSE_STORY_MODE_MODAL);
}
