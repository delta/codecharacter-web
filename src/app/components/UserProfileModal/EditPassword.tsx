import * as styles1 from 'app/styles/Authentication.module.css';
import * as styles from 'app/styles/UserProfileModal.module.css';
import { InputName } from 'app/types/UserProfileModal';
import * as EditPasswordInterfaces from 'app/types/UserProfileModal/EditPassword';
import classnames from 'classnames';
import * as React from 'react';
import { Row } from 'react-bootstrap';

export class EditPassword extends React.Component<EditPasswordInterfaces.Props, {}> {
  public render() {
    const { handleEditPassword, onInputChange } = this.props;
    const { editPasswordRef } = this.props;
    const { oldPassword, password, repeatPassword } = this.props;
    return (
      <div className="col-5">
        <Row className={classnames('mb-3')}>
          <div className={classnames('col-sm-12', styles.form)}>
            <div
              className={classnames('text-dark', styles.formHeading)}
              style={{
                marginBottom: '5%',
                marginLeft: '-1px',
                textAlign: 'center',
              }}
            >
              User Credentials
            </div>
            <form
              className={'editpasswordForm'}
              noValidate
              ref={editPasswordRef}
              onSubmit={handleEditPassword}
            >
              <div className="form-row">
                <div className="col mb-3">
                  <div className="input-group">
                    <label className="labeltext">Old Password</label>
                    <div className={classnames('input-group-prepend', styles['inputGroup'])}>
                      <input
                        type="password"
                        className={classnames('form-control', styles1['login-input'])}
                        id="editValidationOldPassword"
                        aria-describedby="inputGroupPrepend"
                        minLength={5}
                        value={oldPassword}
                        onChange={(e) => onInputChange(InputName.oldPassword, e.target.value)}
                        required
                      />
                      <div className={classnames('invalid-feedback', styles1['login-error'])}>
                        Password should have minimum 5 characters.
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-3">
                  <div className="input-group">
                    <label className="labeltext">New Password</label>
                    <div className={classnames('input-group-prepend', styles['inputGroup'])}>
                      <input
                        type="password"
                        className={classnames('form-control', styles1['login-input'])}
                        id="editValidationPassword"
                        aria-describedby="inputGroupPrepend"
                        minLength={5}
                        value={password}
                        onChange={(e) => onInputChange(InputName.password, e.target.value)}
                        required
                      />
                      <div className={classnames('invalid-feedback', styles1['login-error'])}>
                        Password should have minimum 5 characters.
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col mb-3">
                  <div className="input-group">
                    <label className="labeltext">Confirm Password</label>
                    <div className={classnames('input-group-prepend', styles['passwordInput'])}>
                      <input
                        type="password"
                        className={classnames('form-control', styles1['login-input'])}
                        id="editValidationRepeatPassword"
                        aria-describedby="inputGroupPrepend"
                        minLength={5}
                        value={repeatPassword}
                        onChange={(e) => onInputChange(InputName.repeatPassword, e.target.value)}
                        required
                      />
                      <div className={classnames('invalid-feedback', styles1['login-error'])}>
                        Passwords should match.
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div className="form-row">
                <div className="col-12 text-center">
                  <button
                    className={classnames(
                      'btn btn-success',
                      styles1.loginButton,
                      styles.loginButton,
                    )}
                    type="submit"
                  >
                    Save Changes
                  </button>
                </div>
              </div>
            </form>
          </div>
        </Row>
      </div>
    );
  }
}
