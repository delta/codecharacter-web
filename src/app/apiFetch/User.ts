/* tslint:disable:no-console*/
import {
  HeadReqType,
  headResponseWrapper,
  jsonResponseWrapper,
  setRequestHeaders,
} from 'app/apiFetch/utils';
import * as UserInterfaces from 'app/types/User';
import { API_BASE_URL } from '../../config/config';

export const activateUser = (body: UserInterfaces.ActivateUser) => {
  return fetch(`${API_BASE_URL}user/activate`, {
    body: JSON.stringify(body),
    headers: {
      'Content-Type': 'application/json',
    },
    method: 'POST',
  })
    .then((response) => {
      return headResponseWrapper(response, HeadReqType.USERNAME);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const userLogin = (body: UserInterfaces.Login) => {
  const formBody: string[] = [];
  Object.keys(body).forEach((key) =>
    // @ts-ignore
    formBody.push(`${encodeURIComponent(key)}=${encodeURIComponent(body[key])}`),
  );
  const encodedString = formBody.join('&');

  return fetch(`${API_BASE_URL}login`, {
    body: encodedString,
    credentials: 'include',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
    },
    method: 'POST',
  })
    .then((response) => {
      return jsonResponseWrapper(response);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const userLogout = () => {
  return fetch(`${API_BASE_URL}logout`, {
    credentials: 'include',
    method: 'POST',
  })
    .then((response) => {
      return headResponseWrapper(response, HeadReqType.OTHERS);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const userRegister = (body: UserInterfaces.Register) => {
  return fetch(`${API_BASE_URL}user/`, {
    body: JSON.stringify(body),
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'POST',
  })
    .then((response) => {
      return headResponseWrapper(response, HeadReqType.USERNAME);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const userEditProfile = (body: UserInterfaces.EditUserDetails) => {
  return fetch(`${API_BASE_URL}user`, {
    body: JSON.stringify(body),
    credentials: 'include',
    headers: setRequestHeaders(),
    method: 'PATCH',
  })
    .then((response) => {
      return headResponseWrapper(response, HeadReqType.PROFILE);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const userEditPassword = (body: UserInterfaces.EditUserPassword) => {
  const requestPayLoad = {
    newPassword: body.password,
    oldPassword: body.oldPassword,
  };
  return fetch(`${API_BASE_URL}user/password`, {
    body: JSON.stringify(requestPayLoad),
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    method: 'PATCH',
  })
    .then((response) => {
      return headResponseWrapper(response, HeadReqType.PASSWORD);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const userGetDetails = () => {
  return fetch(`${API_BASE_URL}user`, {
    credentials: 'include',
    method: 'GET',
  })
    .then((response) => {
      return jsonResponseWrapper(response);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const checkEmailExists = (email: string) => {
  const URL = `${API_BASE_URL}user/email/${encodeURIComponent(email)}`;
  return fetch(URL, {
    method: 'HEAD',
  })
    .then((response) => {
      return headResponseWrapper(response, HeadReqType.EMAIL);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const checkUsernameExists = (username: string) => {
  const URL = `${API_BASE_URL}user/username/${encodeURIComponent(username)}`;
  return fetch(URL, {
    method: 'HEAD',
  })
    .then((response) => {
      return headResponseWrapper(response, HeadReqType.USERNAME);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};

export const userGetQuestStatus = () => {
  console.log('Inside apiFetch');
  return fetch(`${API_BASE_URL}user/quest-status`, {
    credentials: 'include',
    method: 'GET',
  })
    .then((response) => {
      return jsonResponseWrapper(response);
    })
    .then((data) => {
      return data;
    })
    .catch((error) => {
      console.error(error);
    });
};
