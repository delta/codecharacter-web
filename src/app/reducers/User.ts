import { UserActions } from 'app/actions';
import * as UserInterfaces from 'app/types/User';

export enum ErrorMessage {
  FORBIDDEN = 'Forbidden',
  UNAUTHORISED = 'Unauthorised',
  INTERNAL_SERVER_ERROR = 'Internal Server Error',
  METHOD_NOT_SUPPORTED = "Request method 'POST' not supported",
  NULL = '',
}

const userStoreIntialState: UserInterfaces.UserStoreState = {
  avatar: '',
  college: '',
  country: 'IN',
  current_level: 1,
  current_stars: 0,
  email: '',
  errorMessage: '',
  fullName: '',
  isAuthenticationOpen: true,
  isFirstLogin: true,
  isLoggedIn: false,
  isLoginLoading: false,
  isNotificationPresent: false,
  isSocketPresent: false,
  isStoryModeModalOpen: false,
  isUserProfileModalOpen: false,
  notification: '',
  ratings: [],
  socketMessage: '',
  storyModeModalLevel: 0,
  userId: 0,
  userType: UserInterfaces.UserType.STUDENT,
  username: '',
};

export const userReducer = (
  state = userStoreIntialState,
  action: UserInterfaces.UserStoreAction,
) => {
  switch (action.type) {
    case UserActions.Type.UPDATE_USER_DETAILS: {
      const {
        country,
        email,
        isLoggedIn,
        username,
        fullName,
        userId,
        isUserProfileModalOpen,
        errorMessage,
        avatar,
        college,
        userType,
        isFirstLogin,
      } = action.payload.userDetails;

      let isAuthenticationOpen = state.isAuthenticationOpen;

      if (!isLoggedIn && state.isLoggedIn) {
        isAuthenticationOpen = true;
      }

      if (isLoggedIn && !state.isLoggedIn) {
        isAuthenticationOpen = false;
      }

      return {
        ...state,
        isAuthenticationOpen,
        avatar: avatar !== undefined ? avatar : state.avatar,
        college: college !== undefined ? college : state.college,
        country: country !== undefined ? country : state.country,
        email: email !== undefined ? email : state.email,
        errorMessage: errorMessage !== undefined ? errorMessage : state.errorMessage,
        fullName: fullName !== undefined ? fullName : state.fullName,
        isFirstLogin: isFirstLogin !== undefined ? isFirstLogin : state.isFirstLogin,
        isLoggedIn: isLoggedIn !== undefined ? isLoggedIn : state.isLoggedIn,
        isUserProfileModalOpen:
          isUserProfileModalOpen !== undefined
            ? isUserProfileModalOpen
            : state.isUserProfileModalOpen,
        userId: userId !== undefined ? userId : state.userId,
        userType: userType !== undefined ? userType : state.userType,
        username: username !== undefined ? username : state.username,
      };
    }

    case UserActions.Type.UPDATE_ERROR_MESSAGE: {
      let errorMessage = action.payload.errorMessage;
      switch (action.payload.errorMessage) {
        case ErrorMessage.FORBIDDEN:
        case ErrorMessage.UNAUTHORISED:
        case ErrorMessage.INTERNAL_SERVER_ERROR:
        case ErrorMessage.METHOD_NOT_SUPPORTED:
          errorMessage = ErrorMessage.NULL;
          break;
      }
      return {
        ...state,
        errorMessage: errorMessage ? errorMessage : '',
      };
    }
    case UserActions.Type.TOGGLE_USER_PROFILE_MODAL: {
      return {
        ...state,
        isUserProfileModalOpen: action.payload.isUserProfileModalOpen,
      };
    }
    case UserActions.Type.SET_IS_AUTHENTICATION_OPEN: {
      return {
        ...state,
        isAuthenticationOpen: action.payload.isAuthenticationOpen,
      };
    }
    case UserActions.Type.SET_IS_LOGIN_LOADING: {
      return {
        ...state,
        isLoginLoading: action.payload.isLoginLoading,
      };
    }
    case UserActions.Type.TOGGLE_IS_SOCKET_PRESENT: {
      return {
        ...state,
        isSocketPresent: !state.isSocketPresent,
      };
    }
    case UserActions.Type.TOGGLE_IS_NOTIFICATION_PRESENT: {
      return {
        ...state,
        isNotificationPresent: !state.isNotificationPresent,
      };
    }
    case UserActions.Type.OPEN_STORY_MODE_MODAL: {
      return {
        ...state,
        isStoryModeModalOpen: true,
        storyModeModalLevel: action.payload.level,
      };
    }
    case UserActions.Type.CLOSE_STORY_MODE_MODAL: {
      return {
        ...state,
        isStoryModeModalOpen: false,
        storyModeModalLevel: -1,
      };
    }
    case UserActions.Type.UPDATE_SOCKET_MESSAGE: {
      return {
        ...state,
        socketMessage: action.payload.socketMessage,
      };
    }
    case UserActions.Type.UPDATE_NOTIFICATION: {
      return {
        ...state,
        notification: action.payload.notification,
      };
    }

    case UserActions.Type.RESET_USER_STATE: {
      return {
        ...userStoreIntialState,
      };
    }

    case UserActions.Type.UPDATE_USER_QUESTSTATUS: {
      return {
        ...state,
        ratings: action.payload.ratings,
      };
    }

    case UserActions.Type.SET_CURRENT_LEVEL: {
      return {
        ...state,
        current_level: action.payload.level,
        current_stars: action.payload.stars,
      };
    }

    default:
      return state;
  }
};
