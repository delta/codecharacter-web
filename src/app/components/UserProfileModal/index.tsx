import { EditPassword } from 'app/components/UserProfileModal/EditPassword';
import { EditProfile } from 'app/components/UserProfileModal/EditProfile';
import * as styles from 'app/styles/UserProfileModal.module.css';
import { AvatarId } from 'app/types/Authentication/Register';
import * as UserProfileInterfaces from 'app/types/UserProfileModal';
import classnames from 'classnames';
import * as React from 'react';
import { Grid, Row } from 'react-bootstrap';
// tslint:disable-next-line
import ReactFlagsSelect from 'react-flags-select';
import { Redirect } from 'react-router-dom';

export class UserProfileModal extends React.Component<
  UserProfileInterfaces.Props,
  UserProfileInterfaces.State
> {
  private editProfileRef = React.createRef<HTMLFormElement>();
  private editPasswordRef = React.createRef<HTMLFormElement>();
  private reactFlagRef = React.createRef<ReactFlagsSelect>();

  constructor(props: UserProfileInterfaces.Props) {
    super(props);
    const { userDetails } = this.props;
    this.state = {
      avatar: userDetails.avatar,
      country: userDetails.country,
      fullName: userDetails.fullName,
      isPasswordPage: true,
      oldPassword: '',
      password: '',
      repeatPassword: '',
      username: userDetails.username,
    };
    this.props.getUserDetails();
  }

  public render() {
    const {
      fullName,
      username,
      country,
      oldPassword,
      password,
      repeatPassword,
      avatar,
    } = this.state;
    const { userDetails } = this.props;
    return (
      <Grid fluid={true} className={classnames(styles.UserEdit)}>
        <Row
          className={
            this.state.isPasswordPage
              ? classnames(styles.editProfileElement)
              : classnames(styles.editPasswordElement)
          }
        >
          {this.state.isPasswordPage ? (
            <EditProfile
              handleEditProfile={this.handleEditProfile}
              onInputChange={this.onInputChange}
              editProfileRef={this.editProfileRef}
              reactFlagRef={this.reactFlagRef}
              username={username}
              fullName={fullName}
              userDetails={userDetails}
              country={country}
              avatar={avatar}
            />
          ) : (
            <EditPassword
              handleEditPassword={this.handleEditPassword}
              onInputChange={this.onInputChange}
              editPasswordRef={this.editPasswordRef}
              oldPassword={oldPassword}
              password={password}
              repeatPassword={repeatPassword}
              userDetails={userDetails}
            />
          )}
        </Row>
        <Row>
          <a
            className={
              this.state.isPasswordPage
                ? classnames('labeltext', styles.loginPageLink)
                : classnames('labeltext', styles.passwordPageLink)
            }
            onClick={() => {
              this.setState((prevState) => ({
                isPasswordPage: !prevState.isPasswordPage,
              }));
            }}
          >
            {this.state.isPasswordPage ? 'Want to change Credentials?' : 'Want to change Info?'}
          </a>
        </Row>
        {this.state.isPasswordPage ? (
          <Row>
            {this.props.userDetails.errorMessage === 'Unauthorised' ? (
              <Redirect to="login" />
            ) : (
              <p>{''}</p>
            )}
          </Row>
        ) : null}
      </Grid>
    );
  }

  private handleEditProfile = (event: React.FormEvent<HTMLFormElement>) => {
    const { editUserProfile } = this.props;
    const { country, fullName, username, avatar } = this.state;
    const form = this.editProfileRef.current;
    event.preventDefault();

    if (form) {
      if (form.checkValidity()) {
        editUserProfile({
          country,
          fullName,
          username,
          // @ts-ignore
          avatarId: AvatarId[avatar],
        });
      }
      form.classList.add('was-validated');
    }
  };

  private handleEditPassword = (event: React.FormEvent<HTMLFormElement>) => {
    const { editUserPassword } = this.props;
    const { oldPassword, repeatPassword, password } = this.state;
    const form = this.editPasswordRef.current;
    event.preventDefault();

    if (form) {
      if (form.checkValidity()) {
        if (password === repeatPassword) {
          editUserPassword({
            oldPassword,
            password,
          });
        }
      }
      form.classList.add('was-validated');
    }
  };

  private onInputChange = (inputName: UserProfileInterfaces.InputName, value: string) => {
    if (inputName === UserProfileInterfaces.InputName.username) {
      const { checkUsernameExists } = this.props;
      checkUsernameExists(value);
    }
    this.setState(this.updateInputState(inputName, value));
  };

  private updateInputState = <T extends string>(
    key: keyof UserProfileInterfaces.State,
    value: T,
  ) => (prevState: UserProfileInterfaces.State): UserProfileInterfaces.State => ({
    ...prevState,
    [key]: value,
  });
}
