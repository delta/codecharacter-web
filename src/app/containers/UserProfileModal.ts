import { UserActions } from 'app/actions';
import { UserProfileModal } from 'app/components/UserProfileModal';
import { RootState } from 'app/reducers';
import * as UserProfileInterfaces from 'app/types/UserProfileModal';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    isUserProfileModalOpen: rootState.user.isUserProfileModalOpen,
    userDetails: { ...rootState.user },
  };
};

const userProfileModalContainer = connect<
  UserProfileInterfaces.StateProps,
  UserProfileInterfaces.DispatchProps,
  {}
>(mapStateToProps, {
  checkEmailExists: UserActions.checkEmailExists,
  checkUsernameExists: UserActions.checkUsernameExists,
  editUserPassword: UserActions.editUserPassword,
  editUserProfile: UserActions.editUserProfile,
  getUserDetails: UserActions.getUserDetails,
  toggleUserProfileModal: UserActions.toggleUserProfileModal,
})(UserProfileModal);

export default userProfileModalContainer;
