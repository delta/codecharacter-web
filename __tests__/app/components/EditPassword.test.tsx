import { EditPassword } from 'app/components/UserProfileModal/EditPassword';
import { UserType } from 'app/types/User';
import { InputName } from 'app/types/UserProfileModal';
import { mount } from 'enzyme';
import * as React from 'react';

describe('EditPassword Component', () => {
  const wrapper = mount(
    <EditPassword
      handleEditPassword={(e: React.FormEvent<HTMLFormElement>) => {
        return;
      }}
      onInputChange={(inputName: InputName, value: string) => {
        return;
      }}
      editPasswordRef={React.createRef<HTMLFormElement>()}
      oldPassword={'oldPassword'}
      password={'password'}
      repeatPassword={'repeatPassword'}
      userDetails={{
        avatar: '',
        college: '',
        country: 'IN',
        current_level: 1,
        current_stars: 1,
        email: '',
        errorMessage: '',
        fullName: '',
        isAuthenticationOpen: true,
        isFirstLogin: true,
        isLoggedIn: false,
        isLoginLoading: false,
        isNotificationPresent: false,
        isSocketPresent: false,
        isStoryModeModalOpen: false,
        isUserProfileModalOpen: false,
        notification: '',
        ratings: [],
        socketMessage: '',
        storyModeModalLevel: 0,
        userId: 0,
        userType: UserType.STUDENT,
        username: '',
      }}
    />,
  );

  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
