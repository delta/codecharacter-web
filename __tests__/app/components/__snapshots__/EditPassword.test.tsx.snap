// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPassword Component Should render 1`] = `
<EditPassword
  editPasswordRef={
    Object {
      "current": <form
        class="editpasswordForm"
        novalidate=""
      >
        <div
          class="form-row"
        >
          <div
            class="col mb-3"
          >
            <div
              class=""
            >
              <label
                class="labeltext"
              >
                Old Password
              </label>
              <div
                class="input-group-prepend"
              >
                <input
                  aria-describedby="inputGroupPrepend"
                  class="form-control"
                  id="editValidationOldPassword"
                  minlength="5"
                  required=""
                  type="password"
                  value="oldPassword"
                />
                <div
                  class="invalid-feedback"
                >
                  Password should have minimum 5 characters.
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="form-row"
        >
          <div
            class="col mb-3"
          >
            <div
              class=""
            >
              <label
                class="labeltext"
              >
                New Password
              </label>
              <div
                class="input-group-prepend"
              >
                <input
                  aria-describedby="inputGroupPrepend"
                  class="form-control"
                  id="editValidationPassword"
                  minlength="5"
                  required=""
                  type="password"
                  value="password"
                />
                <div
                  class="invalid-feedback"
                >
                  Password should have minimum 5 characters.
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="form-row"
        >
          <div
            class="col mb-3"
          >
            <div
              class=""
            >
              <label
                class="labeltext"
              >
                Confirm Password
              </label>
              <div
                class="input-group-prepend"
              >
                <input
                  aria-describedby="inputGroupPrepend"
                  class="form-control"
                  id="editValidationRepeatPassword"
                  minlength="5"
                  required=""
                  type="password"
                  value="repeatPassword"
                />
                <div
                  class="invalid-feedback"
                >
                  Passwords should match.
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="form-row"
        >
          <div
            class="col-12 text-center"
          >
            <button
              class="btn btn-success"
              type="submit"
            >
              Save Changes
            </button>
          </div>
        </div>
      </form>,
    }
  }
  handleEditPassword={[Function]}
  oldPassword="oldPassword"
  onInputChange={[Function]}
  password="password"
  repeatPassword="repeatPassword"
  userDetails={
    Object {
      "avatar": "",
      "college": "",
      "country": "IN",
      "current_level": 1,
      "current_stars": 1,
      "email": "",
      "errorMessage": "",
      "fullName": "",
      "isAuthenticationOpen": true,
      "isFirstLogin": true,
      "isLoggedIn": false,
      "isLoginLoading": false,
      "isNotificationPresent": false,
      "isSocketPresent": false,
      "isStoryModeModalOpen": false,
      "isUserProfileModalOpen": false,
      "notification": "",
      "ratings": Array [],
      "socketMessage": "",
      "storyModeModalLevel": 0,
      "userId": 0,
      "userType": "STUDENT",
      "username": "",
    }
  }
>
  <div
    className="col-5"
  >
    <Row
      bsClass="row"
      className="mb-3"
      componentClass="div"
    >
      <div
        className="mb-3 row"
      >
        <div
          className="col-sm-12"
        >
          <div
            className="text-dark"
            style={
              Object {
                "marginBottom": "5%",
                "marginLeft": "-1px",
                "textAlign": "center",
              }
            }
          >
            User Credentials
          </div>
          <form
            className="editpasswordForm"
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="form-row"
            >
              <div
                className="col mb-3"
              >
                <div
                  className=""
                >
                  <label
                    className="labeltext"
                  >
                    Old Password
                  </label>
                  <div
                    className="input-group-prepend"
                  >
                    <input
                      aria-describedby="inputGroupPrepend"
                      className="form-control"
                      id="editValidationOldPassword"
                      minLength={5}
                      onChange={[Function]}
                      required={true}
                      type="password"
                      value="oldPassword"
                    />
                    <div
                      className="invalid-feedback"
                    >
                      Password should have minimum 5 characters.
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-row"
            >
              <div
                className="col mb-3"
              >
                <div
                  className=""
                >
                  <label
                    className="labeltext"
                  >
                    New Password
                  </label>
                  <div
                    className="input-group-prepend"
                  >
                    <input
                      aria-describedby="inputGroupPrepend"
                      className="form-control"
                      id="editValidationPassword"
                      minLength={5}
                      onChange={[Function]}
                      required={true}
                      type="password"
                      value="password"
                    />
                    <div
                      className="invalid-feedback"
                    >
                      Password should have minimum 5 characters.
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-row"
            >
              <div
                className="col mb-3"
              >
                <div
                  className=""
                >
                  <label
                    className="labeltext"
                  >
                    Confirm Password
                  </label>
                  <div
                    className="input-group-prepend"
                  >
                    <input
                      aria-describedby="inputGroupPrepend"
                      className="form-control"
                      id="editValidationRepeatPassword"
                      minLength={5}
                      onChange={[Function]}
                      required={true}
                      type="password"
                      value="repeatPassword"
                    />
                    <div
                      className="invalid-feedback"
                    >
                      Passwords should match.
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-row"
            >
              <div
                className="col-12 text-center"
              >
                <button
                  className="btn btn-success"
                  type="submit"
                >
                  Save Changes
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </Row>
  </div>
</EditPassword>
`;
